src/Microsoft.AspNetCore.Mvc.Razor/Internal/DependencyContextRazorViewEngineOptionsSetup.cs

```diff
@@ -1,115 +0,0 @@
-
-            SetParseOptions(options, compilationOptions);
-            SetCompilationOptions(options, compilationOptions);
-        }
-
-        // Internal for unit testing.
-        protected internal virtual DependencyContextOptions GetCompilationOptions()
-        {
-            if (!string.IsNullOrEmpty(_hostingEnvironment.ApplicationName))
-            {
-                var applicationAssembly = Assembly.Load(new AssemblyName(_hostingEnvironment.ApplicationName));
-                var dependencyContext = DependencyContext.Load(applicationAssembly);
-                if (dependencyContext?.CompilationOptions != null)
-                {
-                    return dependencyContext.CompilationOptions;
-                }
-            }
-
-            return DependencyContextOptions.Default;
-        }
-
-        private static void SetCompilationOptions(
-            RazorViewEngineOptions options,
-            DependencyContextOptions compilationOptions)
-        {
-            var roslynOptions = options.CompilationOptions;
-
-            // Disable 1702 until roslyn turns this off by default
-            roslynOptions = roslynOptions.WithSpecificDiagnosticOptions(
-                new Dictionary<string, ReportDiagnostic>
-                {
-                    {"CS1701", ReportDiagnostic.Suppress}, // Binding redirects
-                    {"CS1702", ReportDiagnostic.Suppress},
-                    {"CS1705", ReportDiagnostic.Suppress}
-                });
-
-            if (compilationOptions.AllowUnsafe.HasValue)
-            {
-                roslynOptions = roslynOptions.WithAllowUnsafe(compilationOptions.AllowUnsafe.Value);
-            }
-
-            if (compilationOptions.Optimize.HasValue)
-            {
-                var optimizationLevel = compilationOptions.Optimize.Value ?
-                    OptimizationLevel.Release :
-                    OptimizationLevel.Debug;
-                roslynOptions = roslynOptions.WithOptimizationLevel(optimizationLevel);
-            }
-
-            if (compilationOptions.WarningsAsErrors.HasValue)
-            {
-                var reportDiagnostic = compilationOptions.WarningsAsErrors.Value ?
-                    ReportDiagnostic.Error :
-                    ReportDiagnostic.Default;
-                roslynOptions = roslynOptions.WithGeneralDiagnosticOption(reportDiagnostic);
-            }
-
-            options.CompilationOptions = roslynOptions;
-        }
-
-        private static void SetParseOptions(
-            RazorViewEngineOptions options,
-            DependencyContextOptions compilationOptions)
-        {
-            var parseOptions = options.ParseOptions;
-            parseOptions = parseOptions.WithPreprocessorSymbols(
-                parseOptions.PreprocessorSymbolNames.Concat(compilationOptions.Defines));
-
-            LanguageVersion languageVersion;
-            if (!string.IsNullOrEmpty(compilationOptions.LanguageVersion) &&
```

Comment:
> Basically if you don't like the fact that a project setting is required to use c# 7.1

I wasn't complaining about that. To me, it makes sense to explicitly tie parsing a Razor file to the version the Razor parser rather than the project's configuration in the default case. I'd be ok if we allowed the user to explicit way to configure `CSharpParseOptions` and they chose to set it to a language version rather than us implicitly changing it.

That said, I'm not super worried about this behavior.